src/components/themed/TransactionListTop.tsx

```diff
@@ -813,6 +747,10 @@ export function TransactionListTop(props: OwnProps) {
   const account = useSelector(state => state.core.account)
   const exchangeRates = useSelector(state => state.exchangeRates)
   const defaultIsoFiat = useSelector(state => state.ui.settings.defaultIsoFiat)
+  const walletStakingState: WalletStakingState = useSelector(
+    // Fallback to a default state using the reducer if the wallet is not found
+    state => state.staking.walletStakingMap[wallet.id] ?? walletStakingStateReducer(undefined, { type: '' } as any)
```

Comment:
Could you do `export defaultWalletStakingState = walletStakingStateReducer(undefined, { type: 'DUMMY_ACTION_PLEASE_IGNORE' })` in `StakingReducer.ts`, and then use that constant? Then we don't have to re-compute this value on every render. Also, we have the `DUMMY_ACTION_PLEASE_IGNORE` type defined for this exact use-case.